frontend/src/react/OpBlockNoteContainer.tsx

```diff
@@ -60,6 +68,26 @@ export default function OpBlockNoteContainer({ inputField, inputText }: OpBlockN
     ];
   };
 
+  useEffect(() => {
+    const observer = new MutationObserver((mutations) => {
+      mutations.forEach((mutation) => {
+        if (mutation.type === 'attributes' &&
+            (mutation.attributeName === 'data-color-mode' ||
+             mutation.attributeName === 'data-light-theme' ||
+             mutation.attributeName === 'data-dark-theme')) {
+          setTheme(detectTheme());
+        }
+      });
+    });
+
+    observer.observe(document.body, {
+      attributes: true,
+      attributeFilter: ['data-color-mode', 'data-light-theme', 'data-dark-theme']
+    });
+
+    return () => observer.disconnect();
+  }, []);
+
```

Comment:
If I understood correctly this allows for real time updating the editor theme based on config changes. This feels unnecessary to me as it's unlikely to affect any users, really. I'd do the check on page load only.

That said I might be overoptimizing or oversimplifying :sweat_smile: